runtime/src/dependency_tracker.rs

```diff
@@ -16,40 +16,40 @@ fn less_than(a: &Option<u64>, b: u64) -> bool {
 }
 
 impl DependencyTracker {
-    /// Acquire the next work sequence number.
-    /// The sequence starts from 0 and increments by 1 each time it is called.
+    /// Acquire the next work id number.
+    /// The work id starts from 0 and increments by 1 each time it is called.
     pub fn declare_work(&self) -> u64 {
-        self.work_sequence
+        self.work_id
             .fetch_add(1, std::sync::atomic::Ordering::SeqCst)
             + 1
     }
 
-    /// Notify all waiting threads that a work has occurred with the given sequence number.
-    /// This function will update the work sequence and notify all waiting threads only if the work
-    /// sequence is greater than the work sequence. Notify a work of sequence number 's' will
-    /// implicitly imply that all work with sequence number less than 's' have been processed.
-    pub fn mark_this_and_all_previous_work_processed(&self, sequence: u64) {
-        let mut work_sequence = self.processed_work_sequence.lock().unwrap();
-        if less_than(&work_sequence, sequence) {
-            *work_sequence = Some(sequence);
+    /// Notify all waiting threads that a work has been processed with the given work id.
+    /// This function will update the processed work id and notify all waiting threads only if the work
+    /// id is greater than the procsessed work id. Notify a work of id number 's' will
+    /// implicitly imply that all work with id number less than 's' have been processed.
```

Comment:
This comment didn't survive the rename, it doesn't make much sense anymore. How about something like this?:

```suggestion
    /// Notify all waiting threads that work has been processed with the given work id.
    /// This function will notify all waiting threads only if they don't have unfinished work as a dependency.
```